### DISCUSSION TITLE
What is a purpose of the nodeField?

### DISCUSSION THREAD

**Discussion by SVilgelm** [nautilus/gateway#125] [Q&A] [answered]
Hello, I was a bit surprised to see "Node" in the schema. After some research I found this code https://github.com/nautilus/gateway/blob/98c7fbe5dd7c13925512279dc96114c64366b8d8/gateway.go#L281-L294 

I see that the `nodeField` is added to the schema by default here: https://github.com/nautilus/gateway/blob/98c7fbe5dd7c13925512279dc96114c64366b8d8/gateway.go#L136

My question is - what is it? Why do we have by default and is there a way to remove?

**Comment by AlecAivazis** [comment, marked as answer]
Hey @SVilgelm! 

The `node` field and corresponding interface is used by the gateway to correlate entities in various backing services. Because of that, it can't really be removed from the gateway without replacing it with some other method of uniquely identifying an entity of a given type. 

You can read more about the `Node` interface and the gateway's approach to schema federation here: https://itnext.io/a-guide-to-graphql-schema-federation-part-1-995b639ac035
